warehouse/organizations/services.py

```diff
@@ -0,0 +1,145 @@
+        """
+        return self.db.query(Organization).get(organization_id)
+
+    def get_organization_by_name(self, name):
+        """
+        Return the organization object corresponding with the given organization name,
+        or None if there is no organization with that name.
+        """
+        organization_id = self.find_organizationid(name)
+        return (
+            None if organization_id is None else self.get_organization(organization_id)
+        )
+
+    def find_organizationid(self, name):
+        """
+        Find the unique organization identifier for the given name or None if there
+        is no organization with the given name.
+        """
+        try:
+            organization = (
+                self.db.query(Organization.id).filter(Organization.name == name).one()
+            )
+        except NoResultFound:
+            return
+
+        return organization.id
+
+    def add_organization(self, name, display_name, orgtype, link_url, description):
+        """
+        Accepts a organization object, and attempts to create an organization with those
+        attributes.
+        """
+        organization = Organization(
+            name=name,
+            display_name=display_name,
+            orgtype=orgtype,
+            link_url=link_url,
+            description=description,
+        )
+        self.db.add(organization)
+        self.db.flush()
+
+        return organization
+
+    def add_catalog_entry(self, name, organization_id):
+        """
+        Adds the organization name to the organization name catalog
+        """
+        organization = self.get_organization(organization_id)
+        catalog_entry = OrganizationNameCatalog(
+            name=name, organization_id=organization.id
+        )
+
+        self.db.add(catalog_entry)
+        self.db.flush()
+
+        return catalog_entry
+
+    def add_organization_role(self, role_name, user_id, organization_id):
+        """
+        Adds the organization role to the specified user and org
+        """
+        organization = self.get_organization(organization_id)
+        role = OrganizationRole(
+            role_name=role_name, user_id=user_id, organization_id=organization.id
+        )
+
+        self.db.add(role)
+        self.db.flush()
+
+        return role
+
+    def approve_organization(self, organization_id):
+        """
+        Performs operations necessary to approve an Organization
+        """
+        organization = self.get_organization(organization_id)
+        organization.is_active = True
+        organization.is_approved = True
+        organization.date_approved = datetime.datetime.now()
+        # self.db.flush()
+
+        return organization
+
+    def decline_organization(self, organization_id):
+        """
+        Performs operations necessary to reject approval of an Organization
+        """
+        organization = self.get_organization(organization_id)
+        organization.is_approved = False
```

Comment:
Record OrganizationEvent for "declined" include approving user id/username in the additional_data.

Reply:
Added in commit 01fe0ed which will be part of #11208: https://github.com/pypa/warehouse/blob/01fe0ed44efd7b7adce16a92ef5b109e7e0baadf/warehouse/admin/views/organizations.py#L143-L147

Reply:
@divbzero will change this to user.id as @ewdurbin mentioned above.

Reply:
Changed to `user.id` in [db0ec20](https://github.com/pypa/warehouse/pull/11184/commits/db0ec20ab3e83d78934b0a6ae3e58b65006eec92).